src/tagged-type.ts

```diff
@@ -165,12 +168,20 @@ const encodeMap = {
   namedDocument: (value: NamedDocument): TaggedRef => ({
     "@ref": { name: value.name, coll: { "@mod": value.coll.name } },
   }),
-  set: (value: Page<any>) => ({
-    // WIP: core does not accept `@set` tagged values, yet, so just unwrap
-    // "@set": { data: encodeMap["array"](value.data), after: value.after },
-    data: encodeMap["array"](value.data),
-    after: value.after,
-  }),
+  set: (value: Page<any>) => {
+    if (value.data === undefined) {
+      // if a Page has no data, then it must still have an 'after' cursor
+      // WIP: core does not accept `@set` tagged values, yet, so just unwrap for now
+      // return { "@set": value.after }
+      return value.after as string;
+    }
```

Comment:
I'm not sure if we should be encoding `Page` at all until core accepts `@set`. It might create more confusion than anything, especially this string case.

If we must encode it, and it will eventually be `{ @set: value.after }`, then this is prob the right choice

Reply:
Ah you are right, the semantics of sending a set to Fauna will change once they start handling it. I agree it will be better to just not handle encoding Pages yet. We can throw an error and tell folks to use the data and after fields directly instead.